packages/arco-lib/src/components/Table/Table.tsx

```diff
@@ -275,12 +276,31 @@ export const Table = implementRuntimeComponent({
     const selectedRows = currentPageData.filter(d =>
       currentChecked.current.includes(d[rowKey])
     );
+    // TODO: Save clickedRow state when rowkey changes, save the UI of clickedRow when turning the page
+    const clickedRow = currentPageData.find(d => d[rowKey] === currentClickedRow.current);
+    if (!clickedRow) currentClickedRow.current = undefined;
     mergeState({
       selectedRowKeys: selectedRows.map(r => r[rowKey]),
       selectedRows,
+      clickedRow,
     });
   }, [currentPageData, mergeState, rowKey]);
 
+  // If there is less data to display than the current page, reset to the first page
+  useEffect(() => {
+    if (useCustomPagination) return;
+    if (currentPageData.length <= Number(pageSize) * (currentPage - 1)) {
```

Comment:
Shouldn't it be `data.length < Number(pageSize) * (currentPage - 1)?`

Reply:
When the server-side paging option is not enabled, currentPageData equals data